### ISSUE TITLE
Exception messages in server log: Failed to revert World Profiler to original

### ISSUE THREAD

**Issue by Ricket** [The-Fireplace-Minecraft-Mods/TickDynamic#4]
ATM2 pack v0.70

TickDynamic-1.11.2-0.5.0

```
[18:00:28] [Aromabackup-Backup/INFO]: [AromaBackup] Server Backup done!
[18:00:28] [Aromabackup-Backup/INFO] [AromaBackup]: Releasing server Thread.
> [18:00:28] [Server thread/INFO] [AromaBackup]: Server Thread released.
[18:01:10] [Server thread/INFO]: [STDERR]: Failed to revert World Profiler to original
[18:01:10] [Server thread/INFO]: [STDERR]: java.lang.ClassCastException: net.minecraft.profiler.Profiler cannot be cast to com.wildex999.tickdynamic.listinject.CustomProfiler
[18:01:10] [Server thread/INFO]: [STDERR]:      at com.wildex999.tickdynamic.WorldEventHandler.onDimensionUnload(WorldEventHandler.java:92)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_1094_WorldEventHandler_onDimensionUnload_Unload.invoke(.dynamic)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.common.DimensionManager.unloadWorlds(DimensionManager.java:355)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:741)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:384)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:624)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:482)
[18:01:10] [Server thread/INFO]: [STDERR]:      at java.lang.Thread.run(Thread.java:745)
[18:01:10] [Server thread/INFO] [FML]: Unloading dimension 10
[18:01:10] [Server thread/INFO]: [STDERR]: Failed to revert World Profiler to original
[18:01:10] [Server thread/INFO]: [STDERR]: java.lang.ClassCastException: net.minecraft.profiler.Profiler cannot be cast to com.wildex999.tickdynamic.listinject.CustomProfiler
[18:01:10] [Server thread/INFO]: [STDERR]:      at com.wildex999.tickdynamic.WorldEventHandler.onDimensionUnload(WorldEventHandler.java:92)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_1094_WorldEventHandler_onDimensionUnload_Unload.invoke(.dynamic)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.common.DimensionManager.unloadWorlds(DimensionManager.java:355)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:741)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:384)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:624)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:482)
[18:01:10] [Server thread/INFO]: [STDERR]:      at java.lang.Thread.run(Thread.java:745)
[18:01:10] [Server thread/INFO] [FML]: Unloading dimension 1
[18:01:10] [Server thread/INFO]: [STDERR]: Failed to revert World Profiler to original
[18:01:10] [Server thread/INFO]: [STDERR]: java.lang.ClassCastException: net.minecraft.profiler.Profiler cannot be cast to com.wildex999.tickdynamic.listinject.CustomProfiler
[18:01:10] [Server thread/INFO]: [STDERR]:      at com.wildex999.tickdynamic.WorldEventHandler.onDimensionUnload(WorldEventHandler.java:92)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_1094_WorldEventHandler_onDimensionUnload_Unload.invoke(.dynamic)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraftforge.common.DimensionManager.unloadWorlds(DimensionManager.java:355)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:741)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:384)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:624)
[18:01:10] [Server thread/INFO]: [STDERR]:      at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:482)
[18:01:10] [Server thread/INFO]: [STDERR]:      at java.lang.Thread.run(Thread.java:745)
[18:01:10] [Server thread/INFO] [FML]: Unloading dimension -1
>
```

**Comment by The-Fireplace** [comment]
Can you enable debug in the config options, reproduce the issue, and
provide me the full fml-server-latest.log, pasted on a site such as
paste.ubuntu.com?

**Comment by Ricket** [comment]
Sure. It happens every time I start the server, at the very end of the server being ready.

I set debug=true and started & stopped the server. Here is the full modlist and the fml-server-latest.log: https://gist.github.com/Ricket/6a55665a64a4302db913a496e161f7e5

**Comment by The-Fireplace** [comment]
Is that the full log? I scrolled to the end, and don't see anything about it stopping or the crash. Still, that crash report in the first post helps, and I will resolve the issue as soon as possible.

**Comment by The-Fireplace** [comment]
Try version 0.5.2. If the issue persists, comment on this issue and I will re-open it.

**Comment by Ricket** [comment]
You're right. I discovered that if you have a gist with a huge file, and then you edit that gist, it truncates the large file. You have to re-paste it for it not to be truncated. Just fixed it, here are direct links to [fml-server-latest.log](https://gist.github.com/Ricket/6a55665a64a4302db913a496e161f7e5/raw/8582e0f4920e0a6c45cfcebf7a07718691d3e162/02-fml-server-latest.log) and the [console output from the server](https://gist.githubusercontent.com/Ricket/6a55665a64a4302db913a496e161f7e5/raw/8582e0f4920e0a6c45cfcebf7a07718691d3e162/03-stdout.txt) for the same run. fml-server-latest doesn't have the exception stack trace, it's only printed to the console.

Just saw your fix to catch the exception. Is the issue then, that the profiler isn't being applied to the dimension prior to the dimension loading? Maybe the dimension loads before the mod has loaded? Is there something I should change about my server, maybe a config I didn't set? I guess what I'm wondering is, how important is this profiler to the operation of tick dynamic, and how do I check if a given dimension has the tick dynamic profiler? (I don't know about the profiler or its purpose)

**Comment by The-Fireplace** [comment]
The custom profiler is an important part of the mod. Without the profiler, TickDynamic doesn't work. And thanks to your server's console output, I can see why the custom profiler isn't getting set in the first place, and it will be fixed in 0.5.3, which should be out in the next few minutes.
